src/MonitoringFunctions/DataService/Providers/DummyDataService.cs

```diff
@@ -14,9 +14,15 @@ public async Task ReportUrlAccessAsync(string monitorName, HttpResponseMessage h
             await Task.Delay(new Random().Next(200, 4000), cancellationToken).ConfigureAwait(false);
         }
 
+        public async Task ReportScriptExecutionAsync(string monitorName, string scriptName, string commandLineArgs, string error, CancellationToken cancellationToken = default)
+        {
+            await Task.Delay(new Random().Next(200, 4000), cancellationToken).ConfigureAwait(false);
+        }
+
         public void Dispose()
         {
             // Do nothing
         }
+
```

Comment:
consider setting an instance boolean to true the first time dispose is called

Reply:
What value would that add?

Reply:
1. public methods could check the state of the bool & not continue running if already disposed

https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose

Reply:
I agree that the proper way to implement dispose is to add a boolean to track the state. But, for this implementation, there are no resources to release. We are implementing `IDisposable` only to comply with the requirements of the interface.

I don't see how adding a boolean into the class and adding an if check that will probably be optimized away when compiled will benefit us. It will just make the code harder to read and maintain.

Leaving the method empty now doesn't mean a long term commitment. If we have some resources to free tomorrow, it will be pretty easy to add the boolean.